platform/source/mbed_retarget.cpp

```diff
@@ -689,6 +775,18 @@ extern "C" ssize_t write(int fildes, const void *buf, size_t length)
     }
 }
 
+#if MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY && DEVICE_SERIAL
+/* Write the contents of a buffer to a serial interface */
+MBED_WEAK int mbed::minimal_console_putc(int c)
+{
+#if MBED_CONF_TARGET_CONSOLE_UART
+    do_serial_init_once();
+    serial_putc(&stdio_uart, c);
+    return c;
```

Comment:
Missing return if not console UART. Probably best to just return `c`, silently dropping output (not that you check for errors). Also will be smallest code size...